ddl/generated_column.go

```diff
@@ -259,38 +259,35 @@ func (c *illegalFunctionChecker) Leave(inNode ast.Node) (node ast.Node, ok bool)
 	return inNode, true
 }
 
-func checkIllegalFn4GeneratedColumn(colName string, expr ast.ExprNode) error {
-	if expr == nil {
-		return nil
-	}
-	var c illegalFunctionChecker
-	expr.Accept(&c)
-	if c.hasIllegalFunc {
-		return ErrGeneratedColumnFunctionIsNotAllowed.GenWithStackByArgs(colName)
-	}
-	if c.hasAggFunc {
-		return ErrInvalidGroupFuncUse
-	}
-	if c.hasRowVal {
-		return ErrGeneratedColumnRowValueIsNotAllowed.GenWithStackByArgs(colName)
-	}
-	return nil
-}
+const (
+	typeColumn = "column"
+	typeIndex  = "index"
+)
 
-func checkIllegalFn4GeneratedIndex(idxName string, expr ast.ExprNode) error {
+func checkIllegalFn4Generated(name, _type string, expr ast.ExprNode) error {
```

Comment:
using a boolean is enough here. 
If you want to use an explicit name, first, the type should be enum instead of string. Second, the name should not start with '_'.

Reply:
done